File: src/main/java/noteedit/utils/JsonUtils.java
package com.gliese.noteedit.utils;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

/**
 * 封装了Jackson的转换方法
 */
public class JsonUtils {
    public static String objectToJson(Object object) {
        ObjectMapper mapper = new ObjectMapper();
        String json = null;
        try {
            json = mapper.writeValueAsString(object);
        } catch (JsonProcessingException e) {
            e.printStackTrace();
        } finally {
            return json;
        }
    }

    public static Object jsonToObject(String json, Class objectClass) {
        ObjectMapper mapper = new ObjectMapper();
        Object object = null;
        try {
            object = mapper.readValue(json, objectClass);
        } catch (JsonProcessingException e) {
            e.printStackTrace();
        } finally {
            return object;
        }
    }
}

File: src/main/java/noteedit/utils/JsonUtils.kt
package noteedit.utils

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import com.sun.org.apache.bcel.internal.classfile.JavaClass
import kotlin.reflect.KClass

/**
 * 封装了Jackson的转换方法
 */
object JsonUtils {
    fun objectToJson(obj: Any?): String? {
        val mapper = ObjectMapper()
        var json: String? = null
        try {
            json = mapper.writeValueAsString(obj)
        } catch (e: JsonProcessingException) {
            e.printStackTrace()
        }
        return json
    }

    inline fun <reified T> jsonToObject(json: String): T? {
        val mapper = ObjectMapper()
        var obj: T? = null
        try {
            obj = mapper.readValue(json, T::class.java)
        } catch (e: JsonProcessingException) {
            e.printStackTrace()
        }
        return obj
    }
}